clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -2686,14 +2701,14 @@ public void testResetNextBatchExpiry() throws Exception {
         inOrder.verify(client, atLeastOnce()).newClientRequest(anyString(), any(), anyLong(), anyBoolean(), anyInt(), any());
         inOrder.verify(client, atLeastOnce()).send(any(), anyLong());
         inOrder.verify(client).poll(eq(0L), anyLong());
-        inOrder.verify(client).poll(eq(accumulator.getDeliveryTimeoutMs()), anyLong());
+        inOrder.verify(client).poll(geq(Long.MAX_VALUE - System.currentTimeMillis()), anyLong());
```

Comment:
We need to use time, instead of System time. We also need to save the time before the 2nd runOnce() call.